adsmp/solr_updater.py

```diff
@@ -466,6 +466,21 @@ def transform_json_record(db_record):
                         db_record["bibcode"], type(links_data), links_data
                     )
                 )
+
+    # Compute doctype scores on the fly
+    if config.get("DOCTYPE_RANKING", False):
+        doctype_rank = config.get("DOCTYPE_RANKING") 
+        unique_ranks = sorted(set(doctype_rank.values()))
+
+        # Map ranks to scores evenly spaced between 0 and 1 (invert: lowest rank gets the highest score)
+        rank_to_score = {rank: 1 - ( i / (len(unique_ranks) - 1)) for i, rank in enumerate(unique_ranks)}
+
+        # Assign scores to each rank
+        doctype_scores = {doctype: rank_to_score[rank] for doctype, rank in doctype_rank.items()}
+
+        if "doctype" in out.keys():
```

Comment:
Something we will need to check in with Jean-Claude about, will it be an issue for the score calc if some records do not have this field? If yes then we should make sure it is set at all times even if it is set to 0.